scipy/stats/_hypotests.py

```diff
@@ -139,6 +141,253 @@ def epps_singleton_2samp(x, y, t=(0.4, 0.8)):
     return Epps_Singleton_2sampResult(w, p)
 
 
+PoissonMeansTestResult = make_dataclass('PoissonMeansTestResult',
+                                        ('statistic', 'pvalue'))
+
+
+def poisson_means_test(count1, nobs1, count2, nobs2, diff=0,
+                       alternative='two-sided'):
+    r"""
+    Calculates the poisson mean test, the "e-test", for the mean difference of
+    two samples that follow a Poisson distribution from descriptive statistics.
+
+    This is a two-sided test for the null hypothesis that two independent
+    samples have identical average (expected) values.
+
+    Let :math:`X_{11},...,X_{1n_1}` and :math:`X_{21},...,X_{2n_2}` be
+    independent samples, from :math:`Poisson(\lambda_1)` and
+    :math:`Poisson(\lambda_2)` distributions. It is well known that :math:`X_1`
+    and :math:`X_2` are independent:
+
+    .. math:: X_1 = \sum_{i=1}^{n_1} X_{1i} \sim Poisson(n_1\lambda_1)
+
+    .. math:: X_2 = \sum_{i=1}^{n_2} X_{2i} \sim Poisson(n_2\lambda_2)
+
+    Let `count1` and `count2` be the observed values of :math:`X_1` and
+    :math:`X_2`, respectively. The problem of interest here is to test
```

Comment:
These are parameters, but in the documentation they are italicized: 
<img width="884" alt="Screen Shot 2021-08-17 at 3 15 12 PM" src="https://user-images.githubusercontent.com/44255917/129807837-29e10b77-d2bb-435b-aa48-df1e1bad9b38.png">

In other documentation they are seen as code formatted, so I propose that this follow that format. (ex: [stats.gamma shows parameters as code format](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.gamma.html?)). Something else that could be done is note that the word "diff" in the courier font relates to the `diff` parameter.


```suggestion
    Let ``count1`` and ``count2`` be the observed values of :math:`X_1` and
    :math:`X_2`, respectively. The problem of interest here is to test
```

Reply:
I've been arguing that parameters (and all "code") should be formatted in teletype for years. But according to the [numpydoc style guide](https://numpydoc.readthedocs.io/en/latest/format.html) it's "correct" for parameters to be in single backticks, and thus italicized.

![image](https://user-images.githubusercontent.com/6570539/129813984-b92fa363-1a8b-45b2-b96c-711a0db4f24a.png)